deployment/src/main/java/io/quarkiverse/logging/splunk/DevServicesLoggingSplunkProcessor.java

```diff
@@ -0,0 +1,188 @@
+            Runnable closeTask = () -> {
+                if (devService != null) {
+                    try {
+                        devService.close();
+                    } catch (Throwable t) {
+                        log.error("Failed to stop microcks", t);
+                    }
+                }
+                first = true;
+                devService = null;
+                capturedDevServiceConfig = null;
+            };
+            closeBuildItem.addCloseTask(closeTask, true);
+        }
+
+        return devService.toBuildItem();
+    }
+
+    private DevServicesResultBuildItem.RunningDevService startContainer(DevServicesLoggingSplunkConfig config,
+            DockerStatusBuildItem dockerStatusBuildItem, LaunchMode launchMode, Optional<Duration> timeout) {
+        if (!isEnabled(config)) {
+            // explicitly disabled
+            log.info("Not starting devservices for Splunk as it has been disabled in the config");
+            return null;
+        }
+
+        if (!dockerStatusBuildItem.isDockerAvailable()) {
+            log.warn("Configure quarkus.log.handler.splunk.url or have a working docker daemon");
+            return null;
+        }
+
+        DockerImageName dockerImageName = DockerImageName.parse(config.imageName.orElse(SPLUNK_LATEST))
+                .asCompatibleSubstituteFor(SPLUNK_LATEST);
+
+        SplunkContainer splunkContainer = new SplunkContainer(dockerImageName, launchMode == DEVELOPMENT);
+
+        // Configure access to host - getting test-port from config or defaulting to 8081.
+        splunkContainer.withAccessToHost(true);
+        Config globalConfig = ConfigProviderResolver.instance().getConfig();
+        int testPort = globalConfig.getValue("quarkus.http.test-port", OptionalInt.class).orElse(8081);
+
+        if (testPort > 0) {
+            Testcontainers.exposeHostPorts(testPort);
+        }
+
+        // Add envs and timeout if provided.
+        splunkContainer.withEnv(config.containerEnv);
+        timeout.ifPresent(splunkContainer::withStartupTimeout);
+
+        splunkContainer.start();
+
+        return new DevServicesResultBuildItem.RunningDevService(SPLUNK, splunkContainer.getContainerId(),
+                splunkContainer::close, getDevServiceExposedConfig(splunkContainer, config));
+    }
+
+    private Map<String, String> getDevServiceExposedConfig(SplunkContainer container,
+            DevServicesLoggingSplunkConfig config) {
+        final Map<String, String> exposedConfig = new HashMap<>();
+        exposedConfig.put(HANDLER_URL_CONFIG_PROP, container.getSplunkHandlerUrl());
+        exposedConfig.put(API_URL_CONFIG_PROP, container.getSplunkApiUrl());
+        exposedConfig.put("quarkus.log.handler.splunk.token", SplunkContainer.HEC_TOKEN);
+        exposedConfig.put("quarkus.log.handler.splunk.disable-certificate-validation", "true");
+        exposedConfig.put("quarkus.log.handler.splunk.enabled", "true");
+        config.plugNamedHandlers.forEach((k, v) -> {
```

Comment:
What is the reason of configuring named handlers ? 
Is it because we need to declare the handlers that uses splunk (and filter out other handlers?)

Reply:
As far as I could see in the recorder, if one uses named handlers, their configuration is completely independent from the "root" handler. The assumption taken here is to have only one Splunk instance, shared between named handlers. But in order to inject the proper configuration for each named handler, the dev service processor needs to have access to a list of names to build the configuration properties to inject. PB the entire configuration is RUNTIME, whilst dev service things should be buildtime. Hence the idea of an extra map in a build time config bean.

But this needs documenting!